src/web/kestraFsProvider.ts

```diff
@@ -153,7 +160,10 @@ export class KestraFS implements vscode.FileSystemProvider {
 			try {
 				await this.getFlowSource(uri);
 			} catch(e) {
-				throw vscode.FileSystemError.NoPermissions("Cannot create files in 'flows' directory as it's a reserved directory for flows");
+				if(e instanceof vscode.FileSystemError && e.code === 'FileNotFound') {
+					throw vscode.FileSystemError.NoPermissions("Cannot create files in 'flows' directory as it's a reserved directory for flows");
+				}
+				throw e;
```

Comment:
Should catch folder creation too as it will be useless

Reply:
Nope, the folder creation will natively be handled as I handled the 401 inside the `callFileApi`. The goal of this try catch is to differentiate the 404 from the 401. 401 should be rethrown directly as it's permission issue (and so perhaps token expiration) while 404 means the flow doesn't exist.

The goal here is to identify when we try to create a file which is in the flows folder. In such case we should not allow it (till we support flow creation from VSCode). The only way to identify such thing is to call Kestra's flow API to check if a flow with such name exists.